### PR TITLE
Rework client to accept a ClientInterface

### PR DESCRIPTION
**Pull request by Itxaka**
Reworks the cleint stuff a bit to accept an interface so we can easily substitute it with our own for tests.

Signature of the Clients changes a little bit but its all internal stuff, nothing exposed.

Also adds more Fake objects and more tests

Part of #165 

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: pkg/clients/clients.go [hunk 1]
```diff
@@ -42,13 +46,62 @@ import (
...
+}
+
 type Clients struct {
-	*clients.Clients
-	Fleet     fleetcontrollers.Interface
-	Elemental elmcontrollers.Interface
-	Events    corev1Typed.EventInterface
-	Rancher   ranchercontrollers.Interface
-	CAPI      capi.Interface
+	sharedControllerFactory controller.SharedControllerFactory
```
Comment (Itxaka) [inline pkg/clients/clients.go:63]: This drops the generic clients.Clients to include only the needed clients.

I dont event think the `Events` interface is used

File: cmd/operator/display/root.go [hunk 2]
```diff
@@ -60,7 +59,7 @@ func displayRun(path string) {
 		os.Exit(0)
 	}()
 
-	b, err := ioutil.ReadFile(path)
+	b, err := os.ReadFile(path)
```
Comment (davidcassany) [inline cmd/operator/display/root.go:62]: Just discovered ioutil.ReadFile is deprecated :+1:

File: pkg/register/register.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func Register(url string, caCert []byte, smbios bool, emulatedTPM bool, emulated
 	if err != nil {
 		if resp != nil {
 			if resp.StatusCode == http.StatusUnauthorized {
-				data, err := ioutil.ReadAll(resp.Body)
+				data, err := io.ReadAll(resp.Body)
```
Comment (davidcassany) [inline pkg/register/register.go:69]: :+1:

**Review by davidcassany — APPROVED** [review]
Very nice! Yes, using interfaces makes our life way easier for testing :+1:

**Pull request merged**